RHIDP-3836 Audit log: Audit log changes for RHDH 1.3

gwynnemonahan opened the pull request:
RHIDP-3836 Audit log: Audit log changes for RHDH 1.3


<!--- If your changes apply to the latest released and/or in-development version of RHDH, open your PR against the `main` branch and cherrypick your PR to any released branches that you want to apply your changes to. --->

**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):**
1.3

**Issue:**
https://issues.redhat.com/browse/RHIDP-3836

**Link to docs preview:**
https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-486/getting-started-rhdh/#ref-audit-log-file-rotation-overview_assembly-audit-log

**Reviews:**
- [x] SME: @ schultzp2020
- [x] QE: @ gustavolira
- [x] Docs review: @ hmanwani-rh
- [ ] Additional review: @mention assignee (by writer)
<!--- SME approval is required to merge a PR unless the changes are made by a subject matter expert. --->
<!--- QE approval is required to merge a PR unless there are no technical changes to the content. --->
<!--- Docs team approval is required for ALL PRs. --->

Not all SMEs and QEs have permissions to give approving reviews. SME and QE reviews are required, but they may not count as "approved reviews" based on doc repo structure. Still have SME and QE review, but note that the red X for "3 approved reviews" may still appear.

**Additional information:**
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, request reviews from all required stakeholders via Slack GitHub, or Jira. --->

schultzp2020 commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,58 @@
+
+Logging to a rotating file in {product} is helpful for persistent storage for persistent audit logs.
+
+{product-short} supports audit logging for both the catalog and scaffolder. Audit logs can be identified by the `isAuditLog: true` field.
```
There are more plugins that are supported besides catalog and scaffolder.

gwynnemonahan replied:
OK. 

https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/audit-log.md#audit-logging-for-backstage-showcase says: 'Backstage Showcase supports audit logging for both the catalog and scaffolder. Audit logs can be identified by the isAuditLog: true field." 

Currently, https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/getting_started_with_red_hat_developer_hub/assembly-audit-log walks users through audit logs for OpenShift Container Platform, and only has catalog events https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/getting_started_with_red_hat_developer_hub/assembly-audit-log#ref-audit-log-catalog-events.adoc_assembly-audit-log, and scaffolder events https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html/getting_started_with_red_hat_developer_hub/assembly-audit-log#ref-audit-log-scaffolder-events.adoc_assembly-audit-log

What else is supported, and where in Red Hat product docs are those plugins documented? I can update this accordingly with that information.

gwynnemonahan replied:
Per Slack: update to say: {product-short} supports audit logging.

schultzp2020 commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,58 @@
+Supported `auditLog.rotateFile.frequency` options include:
+
+* `daily`: Rotate daily at 00:00 local time
+* `Xm`: Rotate every X minutes (where X is a number between 0 and 59)
```
```suggestion
* `Xm`: Rotate every `X` minutes (where `X` is a number between 0 and 59)
```

schultzp2020 commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,58 @@
+
+* `daily`: Rotate daily at 00:00 local time
+* `Xm`: Rotate every X minutes (where X is a number between 0 and 59)
+* `Xh`: Rotate every X hours (where X is a number between 0 and 23)
```
```suggestion
* `Xh`: Rotate every `X` hours (where `X` is a number between 0 and 23)
```

schultzp2020 commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,58 @@
+
+Logging to a rotating file in {product} is helpful for persistent storage for persistent audit logs.
+
+{product-short} supports audit logging.
+
```
```suggestion
```

I think I would just delete this to be honest.

schultzp2020 commented:
/lgtm

gwynnemonahan commented:
Deleted multitude of rhdh-bot preview links as a preview link is generated with each commit. Deleted old preview links an in effort to reduce the amount of scrolling to get to relevant comments and checks.

gustavolira commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,56 @@
+
+* Rotate logs at midnight (local system timezone).
+* Log file format: `redhat-developer-hub-audit-%DATE%.log`.
+* Log files stored in `/var/log/redhat-developer-hub/audit`.
```
Log files are stored in `/var/log/redhat-developer-hub/audit`

nickboldt commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+// Module included in the following assembly:
+// assembly-audit-log.adoc
+//
```
```suggestion
// 
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+// assembly-audit-log.adoc
+
+[id="proc-audit-log-file-rotation_{context}"]
+= Configure Audit Log file rotation
```
If this is a procedural module, the heading should ideally begin with a gerund.
```suggestion
= Configuring Audit Log file rotation
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+  rotateFile:
+    logFileDirPath: /custom-path <1>
+----
+<1> An absolute path is recommended.
```
As per IBMSG, avoid word recommended. See the screenshot below:
![image](https://github.com/user-attachments/assets/d866f77a-1206-4862-9ed3-8ffd074d6e4d)

```suggestion
<1> Absolute path to the log file.
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
++
+[NOTE]
+====
+The specified directory will be created automatically if it does not exist.
```
```suggestion
The specified directory is created automatically if it does not exist.
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+    maxFilesOrDays: 14 # Deletes the oldest log when there are more than 14 log files
+----
++
+.Example yaml number of days
```
```suggestion
.Example YAML code for number of days
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+
+. Configure a retention policy by the number of files to keep or by the number of days to retain:
++
+.Example yaml number of files
```
```suggestion
.Example YAML code for number of files
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
++
+[NOTE]
+====
+* If log deletion is enabled, a `.<sha256 hash>-audit.json` is generated in the directory where the logs are to track generated logs. Any log file not contained in the directory is not subject to automatic deletion.
```
I'd suggest adding a context statement explaining the note before the bullet points.

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+* A new `.<sha256 hash>-audit.json` file is generated each time the backend starts, which causes previous audit logs to stop being tracked or deleted, except for those still in use by the current backend.
+====
++
+. Archive and compress rotated logs using gzip:
```
Is it gzip or zip?
```suggestion
. Archive and compress rotated logs using `gzip`:
```

gwynnemonahan replied:
It's gzip as it is compressing a single file, in this case a single rotating file. zip would mean there's a central directory at the end and random access to each entry.

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
+
+. Enable audit logging to a rotating file:
++
+[source, yaml]
```
```suggestion
.Example YAML file to enable audit logging
[source, yaml]
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,109 @@
++
+. Customize the log file location:
++
+[source, yaml]
```
```suggestion
.Example customization of log file location
[source, yaml]
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+//
+
+[id="con-audit-log-file-rotation-overview_{context}"]
+= About audit log file rotation
```
I think this heading is user-friendly, and incorporating product names can also improve SEO. wdyt?
```suggestion
= Audit log file rotation in {product}
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+
+== Log file names
+
+Audit log file names are in the following format: `redhat-developer-hub-audit-%DATE%.log` where `%DATE%` is the format specified in [`auditLog.rotateFile.dateFormat`](#configuring-the-file-rotation-frequency). You can customize file names when you configure file rotation.
```
The cross-reference in this sentence is not rendering properly, as the xref format appears to be incorrect.

hmanwani-rh replied:
```suggestion
Audit log file names are in the following format: 

`redhat-developer-hub-audit-%DATE%.log` 

where `%DATE%` is the format specified in xref:anchor_text[]. You can customize file names when you configure the file rotation.
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+* `test`: Rotate every 1 minute
+* `custom`: Use `dateFormat` to set the rotation frequency (default if frequency is not specified)
+
+If `frequency` is set to `Xh`, `Xm` or `test`, the `dateFormat` setting must be configured in a format that includes the specified time component. Otherwise, the rotation will not work as expected.
```
```suggestion
If `frequency` is set to `Xh`, `Xm` or `test`, the `dateFormat` setting must be configured in a format that includes the specified time component. Otherwise, the rotation might not function as expected.
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+    frequency: '17m'
+----
+
+The `dateFormat` setting configures both the %DATE% in logFileName and the file rotation frequency if `frequency` is set to `custom`. The default format is `YYYY-MM-DD`, meaning daily rotation. Supported values are based on [Moment.js formats](https://momentjs.com/docs/#/displaying/format/).
```
Link format is incorrect.
```suggestion
The `dateFormat` setting configures both the `%DATE%` in `logFileName` and the file rotation frequency if `frequency` is set to `custom`. The default format is `YYYY-MM-DD`, meaning daily rotation. Supported values are based on link:https://momentjs.com/docs/#/displaying/format/[Moment.js formats].
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+
+The `dateFormat` setting configures both the %DATE% in logFileName and the file rotation frequency if `frequency` is set to `custom`. The default format is `YYYY-MM-DD`, meaning daily rotation. Supported values are based on [Moment.js formats](https://momentjs.com/docs/#/displaying/format/).
+
+If `frequency` is set to `custom`, then rotations will take place when the date string, represented in the specified `dateFormat`, changes.
```
```suggestion
If the `frequency` is set to `custom`, then rotations take place when the date string, which is represented in the specified `dateFormat`, changes.
```

hmanwani-rh commented on modules/getting-started/con-audit-log-file-rotation-overview.adoc:
```diff
@@ -0,0 +1,57 @@
+
+If `frequency` is set to `custom`, then rotations will take place when the date string, represented in the specified `dateFormat`, changes.
+
+== Archiving and deleting
```
To be consistent with other headings.
```suggestion
== Archive and delete
```

hmanwani-rh reviewed (changes requested):
@gwynnemonahan Nice content. I've added comments/suggestions below. Please take a look and let me know if you've any questions.

gwynnemonahan commented:
@hmanwani-rh , changes made.

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,111 @@
+// assembly-audit-log.adoc
+
+[id="proc-audit-log-file-rotation_{context}"]
+= Configuring Audit Log file rotation
```
```suggestion
= Configuring audit log file rotation
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,111 @@
+* A new `.<sha256 hash>-audit.json` file is generated each time the backend starts, which causes previous audit logs to stop being tracked or deleted, except for those still in use by the current backend.
+====
++
+. Archive and compress rotated logs using `gzip``:
```
```suggestion
. Archive and compress rotated logs using `gzip`:
```

hmanwani-rh commented on modules/getting-started/proc-audit-log-file-rotation.adoc:
```diff
@@ -0,0 +1,111 @@
++
+. Archive and compress rotated logs using `gzip``:
++
+[source, yaml]
```
```suggestion
.Example of archiving or compressing rotated logs
[source, yaml]
```

hmanwani-rh reviewed (commented):
@gwynnemonahan I have a few minor suggestions for your review. Please take a look at them. The rest looks great!

gwynnemonahan commented:
@hmanwani-rh , changes made.

gwynnemonahan commented:
Rebased 09/24/2024.

schultzp2020 reviewed (approved):
/approve
/lgtm

gwynnemonahan commented:
Rebased end of day 09/26/2024.

themr0c commented:
/cherry-pick release-1.3
